contributor-docs/adrs/adr-008-experimental-components.md

```diff
@@ -0,0 +1,70 @@
+# ADR 008: Lowering the barrier to entry for experimental components
+
+## Status
+
+Proposed
+
+&nbsp;
+
+## Context
+
+#### Recap: Drafts
+
+As we work on new components (or rewrites of old components), we often need to start with a "prototype"[^1] and not include them in the semantically versioned (semver-ed) main bundle.
+
+The approach that has served us well since Dec 2021 has been "drafts" (along with causing a lot of confusion).
+
+[ADR-006](./adr-006-drafts.md) explains our use of drafts, but here are the relevant parts:
+
+- When we are not sure about the API of a component, we put it in a different bundle: "primer/react/drafts"
+- A draft has experimental API, but follows the development conventions of primer/react
+- Components in drafts are accompanied by a design spec and go through the process of accessibility reviews. A component can check multiple boxes in the component lifecycle.
+- The contract with consumers is that you are opting into a component in which we don't have 100% confidence yet.
+
+[^1]: Calling this a prototype and not experimental, because that already means something else in the maturity lifecycle.
+
+&nbsp;
+
+#### Problems with drafts
+
+1. **Incompatible with semver**: Because the expectation with drafts is that we are not 100% confident about the API or structure of a component, we slip in breaking changes to draft components as minor releases to primer/react. On the consumer side, this makes every upgrade, even a patch, a potentially breaking upgrade. So, you are effectively opting out of your semver.
+
+1. **Confusion about maturity**: Because of the generic name, "drafts" causes a lot of confusion, especially with "experimental". It is a prototype or experiment, but NOT in the way the [component lifecycle]([https://primer.style/contribute/component-lifecycle]) defines "Experimental" maturity.
+
+1. **Need for experimental candidates**: As more teams build React components and look to upstream reusable components, it is natural to look at drafts as the place to put them. However, the barrier to entry into drafts isn't low (because we still follow design reviews, dev conventions, API reviews, etc.)
+
+Upstreaming a component that was built in the context of a product requires re-evaluating the baked-in assumptions and most likely a change in the API to make it useful for other contexts.
```

Comment:
I agree with this statement, but it doesn't feel like `candidates` fixes this problem. It seems like a persistent one.

Reply:
I think @pksjce articulated it really well:

> I would want our design team/a11y teams to be involved in it. It could also happen that we can breakdown this large component into smaller parts and have them in primer-react. Either way, a buffer for incoming components with shared ownership between product teams and primer-react is a great idea IMO

We need _a place_ where experimental components live that should not be `@primer/react/*` (because of the semver issues mentioned)

Reply:
Ah let me clarify:

I think line 36 is 100% true, but it belongs in its own section. Maybe like "continued risks" or just a section on "this proposal doesn't solve this problem" or something.

Reply:
I think I'm missing the point, can you say more about this?

Reply:
Let's Slack if I'm still being obtuse 😅

Reply:
Ohhh, nah you're totally right. That part should be under "Incompatible with semver" instead of "Need for experimental components"

Reply:
I was trying to make the point that upstreamed components will go through more API changes than (our current usage of) draft components

Reply:
Hi! 

After collecting feedback from the team, I have updated decision no. 3 which is relevant to this conversation

https://github.com/primer/react/pull/2214#issuecomment-1212252442